lib/keyring_liberator.rb

```diff
@@ -24,9 +24,25 @@ def self.get_keyring_named(name)
       get_all_keyrings.find { |k| k.name == name }
     end
 
+    def self.get_current_keyring(name, cwd)
+      found_by_name = name && get_all_keyrings.find { |k| k.name == name && k.path == cwd.to_s }
+      found_by_name || KeyringLiberator.get_keyring(cwd)
+    end
+
+    def self.get_all_keyrings_named(name)
+      get_all_keyrings.find_all { |k| k.name == name }
+    end
```

Comment:
New preferred method of getting the "current" keyring from the Liberator.

The name/path combo is the first check, but falls back to finding the keyring based on the hashed current directory name.